thalwegPlotsMEDIAN.py

```diff
@@ -0,0 +1,418 @@
+
+def plotSlopeAngelAnalysis (db, avaPathLine, avaPathsz, pointList, cfg, outDir, pathList=[], name1=''):
+    
+    """ 
+        create x-y plot of thalweg using s(distances) and z-coordinates
+
+        Parameters
+        -----------
+        db: pandas dataframe
+            dataframe with data of geometries which should be plotted
+        avaPathLine: string
+            name of Thalweg column in db, with x,y,z coordinates
+        avaPathzs: string
+            name of Thalweg column in db, with s and z coordinates
+        pathList: list with strings
+            list with column names of fitted paths
+        pointList: list of strings
+            list with column names of points which should be plotted on thalweg
+        cfg: configuration File
+            information about: slope angles, resample distance, projection, working Dir
+        outDir: pathlib path or str
+            path to folder where plot shall be saved to
+        name1: str
+            name to be added to plate name to indicate what options are used
+        
+        
+    """
+
+    pathlist = []
+    avalancheDir = pathlib.Path(cfg['MAIN']['avalancheDir'])
+    slope1=cfg['MAIN']['slopeAngle1']
+    slope2=cfg['MAIN']['slopeAngle2']
+    projstr=cfg['MAIN']['projstr']
+    fitOption = cfg['FILTERING']['fit']
+  
+    # loop over all events in dbData
+    for index, row in db.iterrows():
+        
+        avaPath = {'x' : row[avaPathLine].xy[0],
+                   'y' : row[avaPathLine].xy[1],
+                   'z' : row[avaPathsz].xy[1],
+                   's' : row[avaPathsz].xy[0]}
+        
+        fig, axes = plt.subplots(figsize=(8,5))
+        
+        if pathList != []:
+            for path in pathList:
+                if path == 'curveFit_s_z':
+                    label = 'parabolic fit until: ' + str(fitOption)
+                    #label = r'fit: $a \cdot \exp(-b \cdot s^2) + c$'
+                    #label = r'fit: $a \cdot \exp(-b \cdot s) + c$'
+                    plt.plot(row[path].xy[0], row[path].xy[1],  label=label, color = 'blue', alpha = 1, lw=1)
+                else:
+                    plt.plot(row[path].xy[0], row[path].xy[1], '--', color ='blue', lw=1)
+            
+        if 'maxpotsize' in row:
+            label = 'thalweg maxpotsize: ' + str(row['maxpotsize'])
+        else: 
+            label = 'full thalweg'
+        plt.plot(avaPath['s'], avaPath['z'], '--', color='grey', label=label)
+        plt.plot(avaPath['s'][int(row['origID']):int(row['depoID'])], avaPath['z'][int(row['origID']):int(row['depoID'])], label='thalweg section of interest', color='k', lw=1.5)
+        
+        for  point in pointList:
+            
+            if pd.notna(row[point]):
+                pointDict = {'x':[row[point].x], 
+                             'y':[row[point].y]}
+                pointAvapath = gT.findClosestPoint(avaPath['x'], avaPath['y'], pointDict)
```

Comment:
if snapped points are provided to function this should not be needed - but if others then needed